### ISSUE TITLE
The classification results is different while using the same net. No 'bnorm' layer is used here.

### ISSUE THREAD

**Issue by anewlearner** [vlfeat/matconvnet#517]
I trained a net and used the net to do classification.  But the FP and TP/P is different every time. Why does this occurs? 
The code is as follows. Thanks!

testx= bsxfun(@minus, testx, imdb.images.data_mean);
net.layers{end}.type = 'softmax';
for t=1:size(testx,4)
    im = testx(:,:,:,t);
    res = vl_simplenn(net, im) ;
    scores = squeeze(gather(res(end).x)) ;
    [bestScore, best] = max(scores) ;
    predicty(t)=best;
end
ind1=find(testy==2);
ind0=find(testy==1);
P=length(ind1);
FP=length(find(predicty(ind0)~=testy(ind0)));
TP=length(find(predicty(ind1)==testy(ind1)));
[FP,TP*100/P]

**Comment by bazilas** [comment]
in case that your network has batch normalization layers, don't forget to add net.mode='test'.

#479 might be also helpful.

**Comment by anewlearner** [comment]
@bazilas After adding 'net.mode='test' ',The problem still remains.
Maybe I should add more background information. I only changed the last layer of cifar10 to do binary classification. The training data was just prepared as follows:
len=length(y);
set1=ones(1,len);
set1(1,round(3*len/4):end)=3;
data=single(x);
dataMean = mean(data(:,:,:,set1 == 1), 4);
data = bsxfun(@minus, data, dataMean) ;
imdb.images.data = data ;
imdb.images.data_mean = dataMean;
imdb.images.labels=single(y)+1;
imdb.images.set=set1;

Only the mean value was subtracted here. So in the test process,I also subtract the mean.
